tiny_dnn/models/vgg16.h

```diff
@@ -0,0 +1,87 @@
+/*
+    Copyright (c) 2013, Taiga Nomi and the respective contributors
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+
+#include <string>
+#include "tiny_dnn/tiny_dnn.h"
+
+#include "tiny_dnn/util/nn_error.h"
+#include "tiny_dnn/util/parameter_init.h"
+
+namespace tiny_dnn {
+namespace models {
+
+/**
+ * VGG16 architecture for tiny-dnn model zoo.
+ *
+ * Reference:
+ * Very Deep Convolutional Networks for Large-Scale Image Recognition
+ * - (https://arxiv.org/abs/1409.1556)
+ */
+class vgg16 : public tiny_dnn::network<tiny_dnn::sequential> {
+ public:
+  explicit vgg16(const std::string &name = "vgg16", bool include_top = true)
+    : tiny_dnn::network<tiny_dnn::sequential>(name) {
+    using conv     = convolutional_layer;
+    using relu     = relu_layer;
+    using max_pool = max_pooling_layer;
+    using dense    = fully_connected_layer;
+
+    // Block 1
+    *this << conv(224, 224, 3, 3, 3, 64, padding::same) << relu();
+    *this << conv(224, 224, 3, 3, 64, 64, padding::same) << relu();
+    *this << max_pool(224, 224, 64, 2) << relu();
+
```

Comment:
We should be able to do this
```cpp
void create_block(tiny_dnn::network<tiny_dnn::sequential> &net, int in_size, int in_channels, int out_channels) {
  net << conv(in_size, in_size, 3, 3, in_channels, out_channels, padding::same) << relu();
  net << conv(in_size, in_size, 3, 3, out_channels, out_channels, padding::same) << relu();
  net << max_pool(in_size, in_size, out_channels, 2) << relu();
}
```
```cpp
int in_size = 224;
int in_channels = 3;
int out_channels = 64;
for (int i = 0; i < num_blocks; ++i) {
  create_block(*this, in_size, in_channels, out_channels);
  in_channels = out_channels;
  out_channels *= 2;
  in_size /= 2;
} 
```

Reply:
Okay so we intend to allow a little modifications over VGG as well ? Sounds good. 

I was planning this thing for Inception and Resnet architecture since Inception and Residual modular blocks are quite popular. Well let's extend this to VGG as well.

Adding :+1:

Reply:
In this way, I guess you could cover different vggs in a single class and define them programmatically

Reply:
Great!